customComponents/threeStateButton.js

```diff
@@ -0,0 +1,24 @@
+webix.protoUI({
+    name: "buttonWithState",
+    $init: function (config) {
+        let state = config.state || 0;
+        config.label = config.states[state];
+        webix.html.addCss(this.$view, "button-state-" + state);
+
+        this.attachEvent("onItemClick", function () {
+            let currentButtonState = this.config.state;
+            webix.html.removeCss(this.$view, "button-state-" + currentButtonState);
+
+            (currentButtonState < (Object.keys(this.config.states).length - 1)) ? currentButtonState += 1 : currentButtonState = 0;
```

Comment:
Dont use ternary operator for cases like this. Use 'if' instead of ternary operator